packages/core/contracts/oracle/implementation/cross-chain/SinkOracle.sol

```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.6.0;
+
+import "./BeaconOracle.sol";
+
+/**
+ * @title Simple implementation of the OracleInterface that is intended to be deployed on non-Mainnet networks and used
+ * to communicate price request data cross-chain with a Source Oracle on Mainnet. An Admin can request prices from
+ * this oracle, which might ultimately request a price from a Mainnet Source Oracle and eventually the DVM.
+ * @dev Admins capable of making price requests to this contract should be OptimisticOracle contracts. This enables
+ * optimistic price resolution on non-Mainnet networks while also providing ultimate security by the DVM on Mainnet.
+ */
+contract SinkOracle is BeaconOracle {
+    constructor(address _finderAddress) public BeaconOracle(_finderAddress) {}
+
+    function requestPrice(
```

Comment:
Do we need to check the registry to ensure the request is coming from an approved contract (to avoid spam)?

Reply:
Hmm we might want to? But that's why I made `BeaconOracle` inherit MultiRole and just set a `Requester` role, but we could instead just re-use the Registry.